src/components/Forms/Form.js

```diff
@@ -0,0 +1,112 @@
+import React     from 'react'
+import PropTypes from 'prop-types'
+import _         from 'underscore'
+
+class Form extends React.Component {
+  static propTypes = {
+    /** Form html chilren */
+    children    : PropTypes.node,
+    /** HTML form attributes */
+    formProps   : PropTypes.shape({}),
+    /** onSubmit callback will pass in model as parameter */
+    onSubmit    : PropTypes.func,
+    /** errors from server mapped to model names. Will attach serverErrors styling to FormComppnents */
+    serverErrors: PropTypes.shape({})
+  }
+
+  static childContextTypes = {
+    ICFormable: PropTypes.object
+  }
+
+  constructor() {
+    super()
+    this.model          = {}
+    this.formComponents = {}
```

Comment:
since we define `state` directly on the class as an instance variable, we should do `model` and `formComponents ` that way too I think